kardia/abi.go

```diff
@@ -94,6 +137,71 @@ func (ec *Client) DecodeInputData(to string, input string) (*types.FunctionCall,
 	}, nil
 }
 
+// UnpackLog returns a log detail
+func (ec *Client) UnpackLog(log *types.Log) (*types.Log, error) {
+	var a *abi.ABI
+	// check if the to address is staking contract, then we search for an event in staking contract ABI
+	if ec.stakingUtil.ContractAddress.Equal(common.HexToAddress(log.ContractAddress)) {
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>ifElseChain: rewrite if-else to switch statement (gocritic)